models/project/project.go

```diff
@@ -330,3 +330,17 @@ func DeleteProjectByIDCtx(ctx context.Context, id int64) error {
 
 	return updateRepositoryProjectCount(ctx, p.RepoID)
 }
+
+func DeleteProjectByRepoIDCtx(ctx context.Context, repoID int64) error {
+	if _, err := db.GetEngine(ctx).Table("project_issue").Join("INNER", "project", "`project`.id = `project_issue`.project_id").Where("`project`.repo_id = ? ", repoID).Delete(&ProjectIssue{}); err != nil {
+		return err
+	}
+	if _, err := db.GetEngine(ctx).Table("project_board").Join("INNER", "project", "`project`.id = `project_board`.project_id").Where("`project`.repo_id = ? ", repoID).Delete(&Board{}); err != nil {
+		return err
+	}
+	if _, err := db.GetEngine(ctx).Table("project").Where("`project`.repo_id = ? ", repoID).Delete(&Project{}); err != nil {
```

Comment:
```suggestion
	if _, err := db.GetEngine(ctx).Delete(&Project{ RepoID: repoID}); err != nil {
```

Reply:
that does not work

Reply:
Why does `Delete` then take an argument?
Simply because there is no method overloading in Go and `Table` could have not been called?
I thought it acted the same way as for `Find`/ `Exist` to automatically supply the `WHERE` clauses based on the non-default values...